### ISSUE TITLE
Task 4:

### ISSUE THREAD

**Issue by knitemblazor** [shahkalp09/Internship#13]
The problem statement is associated with column name classifier

Sub task 1
Master column names
Sno, date, names, job

Sno,birth date, name, job
1,02/03/96,jack, carpenter
2. 09/03/67, Ruby, professor

Sno,date of birth, full name, occupation 
1,02/03/96,jack, carpenter
2. 09/03/67, Ruby, professor

Sno,date of birth, full name, department 
1,02/03/96,jack, carpenter
2. 09/03/67, Ruby, professor

Sno,date of birth, full name, profession
1,02/03/96,jack, carpenter
2. 09/03/67, Ruby, professor

Pipeline should be like
Predict ("profession") > job
Predict ("department") > job
Predict ("occupation") > job

Sub task 2
,,,, <<<<<< Headerless
1,02/03/96,jack, carpenter
2. 09/03/67, Ruby, professor

Pipeline will be like

Check if the table is headerless
Check for the column type and predict the column name
Predict ("1,2")(enrich the features with column data type etc) > sno

Predict ("02/03/96,09/03/67") > date
Predict ("jack,ruby") > name

Create sample data if required you can check internet for data

**Comment by knitemblazor** [comment]
@shahkalp09

**Comment by shahkalp09** [comment]
sir, I should use GNN for this and the Geometric library ?

**Comment by knitemblazor** [comment]
try with something simpler first then explore the possibilities.

**Comment by knitemblazor** [comment]
complexity 2:

employee past records table

emp_id, emp_name, joining_date, ending_date, date_of_birth, insurance_renewal_date

emp_001, jack, 12/3/21, 12/3/23, 12/4/1996, 12/4/23
emp_002, Ria, 12/3/21, _ (nil), 12/4/1996, 12/4/23

----------------------------------------------------------------------------------------------------------------------

emp_id, emp_name, start_date, end_date, dob, med_insurance_renewal_date

emp_001, jack, 12/3/21, 12/3/23, 12/4/1996, 12/4/23
emp_002, Ria, 12/3/21, _ (nil), 12/4/1996, 12/4/23

predict("start_date") = joining_date
predict("joining_date") = joining_date
 
fyi @shahkalp09

**Comment by shahkalp09** [comment]
Ok sir

**Comment by knitemblazor** [comment]
batch size = 4
label = [1,2,3,4]
output = model([[],[],[],[]])
output =[[],
]

**Comment by knitemblazor** [comment]
creation of a dataloader

class data_loader:
       def init():
           pass
       def len():
            calculates the max data sample <<<( this should be a multiple of your batch size)
       def get item():
            picks up data (its a generator function) rt batch size if batch size is for it will be called 4 times

**Comment by shahkalp09** [comment]
Sir, tried so much for the subtask2, if I solve one error then encounter newer one, still trying to complete that.

**Comment by shahkalp09** [comment]
Sir, able to train the simple model successfully end to end for column classifier, now making it more suitable.

**Comment by shahkalp09** [comment]
sir,do we have to connect today?

**Comment by knitemblazor** [comment]
No will connect tomorrow

**Comment by shahkalp09** [comment]
Sir, I have to do few changes then I will upload it.

**Comment by shahkalp09** [comment]
sir, uploaded the code, but the predicted results have still some problems, model training is working.

**Comment by shahkalp09** [comment]
@knitemblazor sir, done with the tapas finetuning and inference with our dataset, I have uploaded it as of now, as pushing has some errors, please check the code.
